app/services/bulk_upload/sales/year2024/row_parser.rb

```diff
@@ -0,0 +1,1323 @@
+      companybuy: %i[field_13],
+      buylivein: %i[field_14],
+
+      jointpur: %i[field_15],
+      buy1livein: %i[field_36],
+      buy2livein: %i[field_43],
+      hholdcount: %i[field_44],
+      stairbought: %i[field_87],
+      stairowned: %i[field_88],
+      socprevten: %i[field_100],
+      mortgageused: %i[field_103 field_117 field_126],
+
+      uprn: %i[field_22],
+      address_line1: %i[field_23],
+      address_line2: %i[field_24],
+      town_or_city: %i[field_25],
+      county: %i[field_26],
+
+      ethnic_group2: %i[field_40],
+      ethnicbuy2: %i[field_40],
+      nationalbuy2: %i[field_41],
+
+      buy2living: %i[field_70],
+      prevtenbuy2: %i[field_71],
+
+      prevshared: %i[field_84],
+
+      staircasesale: %i[field_89],
+    }
+  end
+
+  def attributes_for_log
+    attributes = {}
+
+    attributes["purchid"] = purchaser_code
+    attributes["saledate"] = saledate
+    attributes["noint"] = 2 if field_17 == 1
+
+    attributes["age1_known"] = age1_known?
+    attributes["age1"] = field_31 if attributes["age1_known"]&.zero? && field_31&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["age2_known"] = age2_known?
+    attributes["age2"] = field_38 if attributes["age2_known"]&.zero? && field_38&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["age3_known"] = age3_known?
+    attributes["age3"] = field_46 if attributes["age3_known"]&.zero? && field_46&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["age4_known"] = age4_known?
+    attributes["age4"] = field_50 if attributes["age4_known"]&.zero? && field_50&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["age5_known"] = age5_known?
+    attributes["age5"] = field_54 if attributes["age5_known"]&.zero? && field_54&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["age6_known"] = age6_known?
+    attributes["age6"] = field_58 if attributes["age6_known"]&.zero? && field_58&.match(/\A\d{1,3}\z|\AR\z/)
+
+    attributes["sex1"] = field_32
+    attributes["sex2"] = field_39
+    attributes["sex3"] = field_47
+    attributes["sex4"] = field_51
+    attributes["sex5"] = field_55
+    attributes["sex6"] = field_59
+
+    attributes["relat2"] = field_37
+    attributes["relat3"] = field_45
+    attributes["relat4"] = field_49
+    attributes["relat5"] = field_53
+    attributes["relat6"] = field_57
+
+    attributes["ecstat1"] = field_35
+    attributes["ecstat2"] = field_42
+    attributes["ecstat3"] = field_48
+    attributes["ecstat4"] = field_52
+    attributes["ecstat5"] = field_56
+    attributes["ecstat6"] = field_60
+
+    attributes["details_known_2"] = details_known?(2)
+    attributes["details_known_3"] = details_known?(3)
+    attributes["details_known_4"] = details_known?(4)
+    attributes["details_known_5"] = details_known?(5)
+    attributes["details_known_6"] = details_known?(6)
+
+    attributes["ethnic_group"] = ethnic_group_from_ethnic
+    attributes["ethnic"] = field_33
+    attributes["national"] = field_34
+
+    attributes["income1nk"] = field_77.present? ? 0 : 1
```

Comment:
Come to think of it, this will also want the change from https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/2152

Reply:
Oooh, good spot! Thanks!